src/main/java/com/example/hop_oasis/service/data/JwtService.java

```diff
@@ -0,0 +1,73 @@
+package com.example.hop_oasis.service.data;
+
+import com.example.hop_oasis.model.User;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.io.Decoders;
+import io.jsonwebtoken.security.Keys;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Service;
+
+import javax.crypto.SecretKey;
+import java.util.Date;
+import java.util.function.Function;
+
+@Service
+public class JwtService {
+    @Value("${special.jwt.value}")
+    private String secretKey;
+    @Value("${special.time.value}")
+    private Long timeValue;
+
+
+    public String extractUsername(String token) {
+        return extractClaim(token, Claims::getSubject);
+    }
+
+    public boolean isValid(String token, UserDetails user) {
+        String userEmail = extractUsername(token);
+        return (userEmail.equals(user.getUsername())) && !isTokenExpired(token);
```

Comment:
remove "user" argument. Token is valid if username is not null, it exists in the DB and token is not expired